default/python/AI/PolicyAI.py

```diff
@@ -122,6 +122,21 @@ def __init__(self, status_only: bool = False):
         self._wanted_ip = self._wanted_for_bureaucracy()
         self._adoptable = self._get_adoptable()
         self._available = set(self._empire.availablePolicies)
+        self._rating_functions = {
+            propaganda: lambda: 20 + self._rate_opinion(propaganda),
+            algo_research: self._rate_algo_research,
+            diversity: self._rate_diversity,
+            artisans: self._rate_artisans,
+            # military policies are mostly chosen by opinion, plus some rule-of-thumb values
+            allied_repair: lambda: self._rate_opinion(charge),  # no effect, unless we have allies...
```

Comment:
For such cases you could use [partilal](https://docs.python.org/3/library/functools.html#functools.partial)
```suggestion
            allied_repair: partial(self._rate_opinion, charge),  # no effect, unless we have allies...
```